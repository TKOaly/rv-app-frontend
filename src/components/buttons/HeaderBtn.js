import './styles/Button.css';
import './styles/HeaderBtn.css';
import Loader from '../loaders/Loader';
import PropTypes from 'prop-types';
import React from 'react';

const HeaderBtn = ({ onClick, children, fill, hover, loader, ...props }) => {
    let className = 'btn';
    fill ? (className += ' headerbtn-fill') : (className += ' headerbtn');
    hover && (fill ? (className += ' headerbtn-fill-hover') : (className += ' headerbtn-hover'));

    return (
        <button {...props} onClick={onClick} className={className}>
            {!loader ? <span className="btnContent">{children}</span> : <Loader />}
        </button>
    );
};

HeaderBtn.propTypes = {
    /** Click handler. */
    onClick: PropTypes.func,
    /** Button fill. If set to `true`, the button's colors will "invert". */
    fill: PropTypes.bool,
    /** Hover effect. If set to `true`, the button will have a hover effect. */
    hover: PropTypes.bool,
    /** Loader effect. If set to `true`, will show a CSS loader instead of button text. */
    loader: PropTypes.bool
};

HeaderBtn.defaultProps = {
    onClick: null,
    fill: false,
    hover: false,
    loader: false
};

export default HeaderBtn;
