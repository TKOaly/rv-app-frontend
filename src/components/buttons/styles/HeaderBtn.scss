@import "vars";

/*
  Header buttons
  */

.headerbtn {
  color: $header-button-bgcolor;
  -webkit-transition: background-color 0.1s; /* Safari */
  transition: background-color 0.1s;
}

.headerbtn-hover:hover {
  color: $header-button-textcolor;
  background-color: $header-button-bgcolor;
  border-color: $header-button-bgcolor;
  cursor: pointer;
}

.headerbtn-fill {
  color: $header-button-textcolor;
  background-color: $header-button-bgcolor;
  border-color: $header-button-bgcolor;
  -webkit-transition: background-color 0.1s; /* Safari */
  transition: background-color 0.1s;
}

.headerbtn-fill-hover:hover {
  color: $header-button-bgcolor;
  background-color: transparent;
  border-color: $header-button-bgcolor;
  cursor: pointer;
}
