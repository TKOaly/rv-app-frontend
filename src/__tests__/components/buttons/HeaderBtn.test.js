import HeaderBtn from '../../../components/buttons/HeaderBtn';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<HeaderBtn/>', () => {
    it('renders without crashing', () => {
        const cmpnt = renderer.create(<HeaderBtn />).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });

});
