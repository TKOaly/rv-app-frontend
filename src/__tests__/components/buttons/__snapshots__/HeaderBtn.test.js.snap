// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBtn/> renders without crashing 1`] = `
<button
  className="btn headerbtn"
  onClick={null}
>
  <span
    className="btnContent"
  />
</button>
`;
